dmci/distributors/pycsw_dist.py

```diff
@@ -0,0 +1,188 @@
+                         'xsi:schemaLocation="http://www.opengis.net/cat/csw/2.0.2 '
+                         'http://schemas.opengis.net/csw/2.0.2/CSW-publication.xsd" '
+                         'service="CSW" version="2.0.2">'
+                         '   <csw:Delete>'
+                         '       <csw:Constraint version="1.1.0">'
+                         '           <ogc:Filter>'
+                         '               <ogc:PropertyIsEqualTo>'
+                         '                   <ogc:PropertyName>apiso:Identifier</ogc:PropertyName>'
+                         '                   <ogc:Literal>%s</ogc:Literal>'
+                         '               </ogc:PropertyIsEqualTo>'
+                         '           </ogc:Filter>'
+                         '       </csw:Constraint>'
+                         '   </csw:Delete>'
+                         '</csw:Transaction>' % self._metadata_id)
+        resp = requests.post(self._conf.csw_service_url, headers=headers, data=xml_as_string)
+        status = self._get_transaction_status(self.TOTAL_DELETED, resp)
+
+        return status
+
+    def _get_transaction_status(self, key, resp):
+        """ Check response status, read response text, and get status (boolean)
+
+        Input
+        =====
+        key : str
+            total_inserted, total_updated or total_deleted
+        resp : requests.models.Response
+            Response object from transaction
+
+        Return
+        ======
+        status : boolean
+            True if the transaction succeeded, otherwise False
+        """
+        if key not in self.STATUS:
+            logger.error("Input key must be '%s', '%s' or '%s'" % (self.TOTAL_INSERTED,
+                                                                   self.TOTAL_UPDATED,
+                                                                   self.TOTAL_DELETED))
+            return False
+        if not type(resp) is requests.models.Response:
+            logger.error("Input argument 'resp' must be instance of requests.models.Response")
+            return False
+        status = False
+        if resp.status_code >= 200 and resp.status_code < 300:
+            status = self._read_response_text(key, resp.text)
+        else:
+            logger.error(resp.text)
+
+        return status
+
+    def _read_response_text(self, key, text):
+        """ Read response xml text and return a dict with results
+
+        Input
+        =====
+        text : str
+            xml representation of the pycsw result
+
+        Return
+        ======
+        dict : status of insert, update and delete
+        """
+        if key not in self.STATUS:
+            logger.error("Input key must be '%s', '%s' or '%s'" % (self.TOTAL_INSERTED,
+                                                                   self.TOTAL_UPDATED,
+                                                                   self.TOTAL_DELETED))
```

Comment:
Just a suggestion for code format. This is much more readable:
```python
logger.error("Input key must be '%s', '%s' or '%s'" % (
    self.TOTAL_INSERTED, self.TOTAL_UPDATED, self.TOTAL_DELETED)
)
```

Reply:
Solved in e0d9e2b758513483ef47ba55dc16eef4f4704ff9